paasta_tools/firewall.py

```diff
@@ -182,19 +188,29 @@ def _yocalhost_rule(port, comment, protocol="tcp"):
     )
 
 
-def _reject_remaining_inbound_traffic_rule(port, protocol="tcp"):
-    """Return an iptables rule denying all other traffic.
-
-    This should eventually be turned into a deny-allow,
-    but is opt-in at the service level for now."""
-    return iptables.Rule(
-        protocol=protocol,
-        src="0.0.0.0/0.0.0.0",
-        dst="0.0.0.0/0.0.0.0",
-        target="REJECT",
-        matches=((protocol, (("dport", (str(port),)),)),),
-        target_parameters=((("reject-with", ("icmp-port-unreachable",))),),
-    )
+def _inbound_traffic_rule(conf, service_name, instance_name, protocol="tcp"):
+    """Return iptables rules for inbound traffic
+
+    If this is set to "reject", this is limited only to traffic from localhost"""
+    policy = conf.get_inbound_firewall()
+    if policy == "reject":
+        port = conf.get_proxy_port(service_name, instance_name)
+        yield iptables.Rule(
+            protocol=protocol,
+            src="0.0.0.0/0.0.0.0",
+            dst="0.0.0.0/0.0.0.0",
+            target="REJECT",
+            matches=((protocol, (("dport", (str(port),)),)),),
+            target_parameters=((("reject-with", ("icmp-port-unreachable",))),),
+        )
+        yield iptables.Rule(
```

Comment:
do you need moar local IPs like 169.254.255.254 169.254.0.1 etc?

Reply:
That depends on whether we want to continue to support smartstack traffic (re, our conversation about using this with envoy). If so, we'll also want those two IPs.

I guess they can't hurt regardless. I'll add them.